statsd/client/udp.py

```diff
@@ -46,5 +46,10 @@ def _send(self, data):
             # No time for love, Dr. Jones!
             pass
 
+    def close(self):
+        if self._sock and hasattr(self._sock, 'close'):
+            self._sock.close()
+        self._sock = None
+
```

Comment:
I think you probably should move this to StatClientBase class and  remove these lines from [`StreamClientBase` class](https://github.com/jsocol/pystatsd/blob/1c90b9fdf322680e2625da659abc2aa5d79b5bff/statsd/client/stream.py#L18-L21)

Reply:
I actually disagree. `self._sock` is an implementation detail used by both the `StreamClientBase` and the `udp.StatsClient` implementation, but there's no requirement that it be true. You could implement this interface in-memory or that output to stderr/stdout, if you wanted to.

I am going to add an unimplemented `close()` method to the base class, though. In the case of an implementation like I mentioned, you would no-op it.

Reply:
> but there's no requirement that it be true.
Good one. If there is no contract like this. Then it should be up to the implementor. 
+1